src/vmm/src/rpc_interface.rs

```diff
@@ -485,6 +514,84 @@ impl RuntimeApiController {
         Ok(())
     }
 
+    /// Updates configuration for the balloon device as described in `balloon_update`.
+    fn latest_balloon_stats(&self) -> result::Result<VmmData, VmmActionError> {
+        if let Some(busdev) = self
+            .vmm
+            .lock()
+            .expect("Poisoned lock")
+            .get_bus_device(DeviceType::Virtio(TYPE_BALLOON), BALLOON_DEV_ID)
+        {
+            let virtio_device = busdev
+                .lock()
+                .expect("Poisoned lock")
+                .as_any()
+                .downcast_ref::<MmioTransport>()
+                // Only MmioTransport implements BusDevice at this point.
+                .expect("Unexpected BusDevice type")
+                .device();
+
+            let latest_stats = virtio_device
+                .lock()
+                .expect("Poisoned lock")
+                .as_mut_any()
+                .downcast_mut::<Balloon>()
+                .unwrap()
+                .latest_stats()
+                .map(|stats| VmmData::BalloonStats(stats.clone()));
+
+            latest_stats.ok_or(VmmActionError::BalloonConfig(
+                BalloonConfigError::StatsNotFound,
+            ))
+        } else {
+            Err(VmmActionError::BalloonConfig(
+                BalloonConfigError::DeviceNotFound,
+            ))
+        }
+    }
+
+    /// Updates configuration for the balloon device as described in `balloon_update`.
+    fn update_balloon_config(&mut self, balloon_update: BalloonUpdateConfig) -> ActionResult {
```

Comment:
same with this: move to `Vmm`

(I know block and net updates are also here, but those should also move to the vmm impl/interface - outside the scope of this PR though)